src/client/components/modals/Organization.js

```diff
@@ -0,0 +1,251 @@
+import React, { useContext, useState } from 'react'
+import { Field, Text, Button, theme } from '@aragon/ui'
+import PropTypes from 'prop-types'
+import { ModalWrapper, DisplayErrors } from './ModalWrapper'
+import { validateDAOAddress } from '../../utils/validation'
+import { shortDAOAddress, fakeIsMember } from '../../utils'
+import { BoxContext } from '../../wrappers/box'
+import { ModalContext } from '../../wrappers/modal'
+import { close } from '../../stateManagers/modal'
+import { isEmpty } from 'lodash'
+import styled from 'styled-components'
+import { useProfileStates } from '../../hooks'
+import {
+  requestedCheckMembership,
+  requestedCheckMembershipReset,
+  requestedCheckMembershipSuccess,
+  requestedCheckMembershipError,
+} from '../../stateManagers/box'
+import {
+  TextInputWithValidation,
+  AnimationLoadingCircle,
+  IconSuccess,
+  IconError,
+} from '../../components/styled-components'
+
+const CheckWrapper = styled.div`
+  height: 146px;
+  display: flex;
+  align-items: center;
+  padding-bottom: 13px;
+  > :first-child {
+    margin-left: 10px;
+    margin-right: 30px;
+  }
+`
+
+const BackToPreviousScreen = ({ ethereumAddress, dispatch }) => (
+  <Text.Block
+    css={`
+      text-align: right;
+      margin-bottom: 0px;
+      text-decoration: underline;
+      cursor: pointer;
+      vertical-align: text-bottom;
+      position: absolute;
+      bottom: 0;
+      right: 0;
+    `}
+    color={theme.accent}
+    onClick={() => dispatch(requestedCheckMembershipReset(ethereumAddress))}
+  >
+    Back to previous screen
+  </Text.Block>
+)
+
+BackToPreviousScreen.propTypes = {
+  ethereumAddress: PropTypes.string.isRequired,
+  dispatch: PropTypes.func.isRequired,
+}
+
+const CheckInfo = ({
+  type,
+  ethereumAddress,
+  address,
+  dispatch,
+  linkBack = false,
+}) => {
+  const checkInfoIcon = {
+    started: AnimationLoadingCircle,
+    success: IconSuccess,
+    error: IconError,
+  }
+  const checkInfoText = {
+    started: `Validating your membership to ${shortDAOAddress(address)}...`,
+    success: `We found ${shortDAOAddress(
+      address
+    )} and confirmed you are a member`,
+    error: `We could not verify your membership to ${shortDAOAddress(address)}`,
+  }
+
+  return (
+    <ModalWrapper>
+      <CheckWrapper>
+        {checkInfoIcon[type]()}
```

Comment:
Why are you calling a function here?

Reply:
Because checkInfoIcon is full of components that need to actually be created before use.

Reply:
Ahh I see I didn't look close enough and realize those were components. 

I try to avoid calling methods inside JSX to avoid unnecessary rerenders. In this scenario, I would make another little component called `CheckStatusIcon`, move it to styled-components directory, and have it take 3 props: `success`, `started`, `error`.

```js
const CheckInfoIcon = ({ starting, success, error }) => {
   if (starting) return <Component1 />
   if (success) return ...
   {... }
}
```

And then on line 84 you can simply do: 

`<CheckInfoIcon { ...type } />` (I think that destructuring should work)

I believe this will reduce # of renders

Reply:
This wasn't quite working for me, but this was:

const Icon = checkInfoIcon[type]
<Icon />

Similar idea I think...